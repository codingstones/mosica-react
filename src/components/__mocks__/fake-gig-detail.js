export const fakeGigDetail = {
  address: "Avda. Cesáreo Alierta, 120, 50013, Zaragoza",
  affiliate_url: null,
  day: "2017-12-16",
  description:
    "Enrique Bunbury ha anunciado el lanzamiento de su nuevo disco de estudio Expectativas. Hará una parada en Zaragoza el próximo 16 de diciembre para presentarlo ante sus paisanos. Este nuevo álbum es la continuación de Palosanto de 2013. La espera ha sido de cuatro años, pero el 8 de septiembre llegarán por fin los dos primeros adelantos,",
  hour: "22:00",
  id: 6533,
  image_url: "http://www.zaragoza.es/cont/paginas/actividades/imagen/pfotoevento4020.png",
  lat_lng: "41.63602275205834,-0.8658168875758695",
  place: "Pabellón Príncipe Felipe",
  price: "",
  schedule: [],
  schedule_id: 9119,
  slug: "concierto-enrique-bunbury",
  title: "Concierto Enrique Bunbury"
}