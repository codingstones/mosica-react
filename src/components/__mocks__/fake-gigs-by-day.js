export const fakeGigsByDay = [
  {
    'day': 'Lunes, 18 de Septiembre',
    'gigs': [
      {
        'id': '123456',
        'title': 'Anarchy in the JS',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'BCN',
        'description': 'Coding Stones GigRow',
        'price': '',
        'image_url': 'http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png',
        'affiliate_url': null,
        'place': 'SCBCN',
        'slug': 'stones',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-09-18',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      },
      {
        'id': '2222222',
        'title': 'Anarchy in the Codemotion',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'BCN',
        'description': 'Another Coding Stones GigRow',
        'price': '',
        'image_url': 'http://www.wtf.es/wtf.png',
        'affiliate_url': null,
        'place': 'Around the world',
        'slug': 'stones',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-12-03',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      },
    ]
  }
]

export const FIRST_DAY_GIG_TEXTS = fakeGigsByDay[0].gigs.map(gig => gig.title)
export const FIRST_GIG = fakeGigsByDay[0].gigs[0]
