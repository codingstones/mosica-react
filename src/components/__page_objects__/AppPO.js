export default class AppPO {
  constructor(wrapper) {
    this.wrapper = wrapper;
  }

  async updateAsync() {
    await flushPromises();
    this.wrapper.update();
  }

  dayTexts() {
    return this.wrapper.find('.day').map((day) => day.text());
  }

  gigRowsFor(dayNumber) {
    return this.wrapper.find('.gigs').at(dayNumber).find('.row').map((row) => row.text())
  }

  text() {
    return this.wrapper.text();
  }
}