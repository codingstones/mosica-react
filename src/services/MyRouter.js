export const MyRouter = (history) => {
  return { navigateToGig }

  function navigateToGig(gigId) {
    history.push("/gig/" + gigId)
  }
}
