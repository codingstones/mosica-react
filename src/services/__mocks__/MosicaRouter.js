const navigateToGigMock = jest.fn()
const RouterMock = { navigateToGig: navigateToGigMock}

const MosicaRouter = (history)=> {
  return RouterMock;
}
module.exports = {
  MosicaRouter
}

