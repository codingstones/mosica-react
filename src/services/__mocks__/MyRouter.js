const navigateToGigMock = jest.fn()
const RouterMock = { navigateToGig: navigateToGigMock}

const MyRouter = (history)=> {
  return RouterMock;
}
module.exports = {
  MyRouter: MyRouter,
}

