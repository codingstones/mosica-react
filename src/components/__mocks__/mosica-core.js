import { fakeGigsByDay } from './fake-gigs-by-day'
const mosicaCore = jest.genMockFromModule('mosica-core')

class GigService {
  retrieveNextGigs() {
    return Promise.resolve(fakeGigsByDay)
  }
}

mosicaCore.GigService = GigService
module.exports = mosicaCore
