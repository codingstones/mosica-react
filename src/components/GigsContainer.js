import React, { Component } from 'react';
import './App.css';
import fetchJsonp from 'fetch-jsonp'
import * as mosicaCore from 'mosica-core'
import { HttpClient } from '../services/HttpClient'
import { GigsDay } from './GigsDay';
import { LoadSpinner } from './LoadSpinner';
import { withRouter } from 'react-router-dom'
import { MyRouter } from '../services/MyRouter';


class GigsContainer extends Component {
  constructor(props) {
    super(props)
    this.state = { gigsByDay: [], isLoading: true }
    this.router = MyRouter(this.props.history)
  }

  async componentDidMount() {
    const gigService = new mosicaCore.GigService(HttpClient(fetchJsonp), new mosicaCore.Matcher())
    const gigsByDay = await gigService.retrieveNextGigs()
    this.setState({ isLoading: false })
    return this.setState({ gigsByDay })
  }

  goToGig = (gigId) => ()=> {
    this.router.navigateToGig(gigId);
  }

  render() {
    return (
      <div>
        <LoadSpinner isLoading={this.state.isLoading}/>
        <GigsDay days={this.state.gigsByDay} onClick={this.goToGig}/>
      </div>
    );
  }
}

export default withRouter(GigsContainer);
export const GigsContainerWithoutRouter = GigsContainer;
