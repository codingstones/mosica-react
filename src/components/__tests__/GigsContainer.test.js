import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { GigsContainerWithoutRouter as GigsContainer } from '../GigsContainer';
import AppPO from '../__page_objects__/GigContainerPO';
import {fakeGigsByDay} from '../__mocks__/fake-gigs-by-day';
import { MyRouter } from '../../services/MyRouter';
jest.mock('../../services/MyRouter');
let navigateToGigMock;
beforeEach(() => {
  navigateToGigMock = MyRouter().navigateToGig
  navigateToGigMock.mockClear()
});
// jest.unmock('mosica-core')

it('renders GigsContainer', () => {
  const component = renderer.create(<GigsContainer />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('When GigsContainer is rendered with async gigs', async () => {

  let app, wrapper;
  beforeEach(async () => {
    wrapper = mount(
      <GigsContainer />
    );
    app = new AppPO(wrapper);
    await app.updateAsync();

  });

  it('shows a concrete gig title', async () => {
    const GIG_TITLE = fakeGigsByDay[0].gigs[1].title;
    expect(app.text()).toContain(GIG_TITLE);
  });

  it('shows days', async () => {
    const DAYS = fakeGigsByDay.map((day)=> day.day);
    expect(app.dayTexts()).toEqual(DAYS);
  });

  it('shows gigs for each day', async () => {
    //IS THIS TEST WORTHY? Maybe easier to unit-test components
    let expected_gigs;
    fakeGigsByDay.map((day, index)=> {
      expected_gigs = day.gigs.map((gig) => gig.title + '-' + gig.place)
      expect(app.gigRowsFor(index)).toEqual(expected_gigs);
    });
  });

  it('navigates to gig detail when click', async () => {
    app.clickFirstGig();
    expect(navigateToGigMock).toHaveBeenCalledWith('123456');
  });

  it('navigates to gig detail when click', async () => {
    app.clickSecondGig();
    expect(navigateToGigMock).toHaveBeenCalledWith('2222222');
  });
});

