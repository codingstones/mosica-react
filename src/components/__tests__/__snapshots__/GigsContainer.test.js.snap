// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GigsContainer 1`] = `
<div>
  <div
    className="spinner"
  >
    <p>
      Loading...
    </p>
  </div>
  <div />
</div>
`;
